src/transformers/trainer_callback.py

```diff
@@ -475,3 +476,58 @@ def on_log(self, args, state, control, logs=None, **kwargs):
         _ = logs.pop("total_flos", None)
         if state.is_local_process_zero:
             print(logs)
+
+
+class EarlyStoppingCallback(TrainerCallback):
+    """
+    An early stopping callback
+
+    Args:
+       early_stopping_patience (:obj:`int`):
+            Use with :obj:`metric_for_best_model` to stop training when the specified metric worsens for
+            :obj:`early_stopping_patience` evaluation calls.
+       early_stopping_threshold(:obj:`float`, `optional`):
+            Use with TrainingAruments :obj:`metric_for_best_model` and :obj:`early_stopping_patience` to denote how
+            much the specified metric must improve to satisfy early stopping conditions. `
+
+    Internal State: early_stopping_patience_counter (:obj:`int`): The number of times validation metrics failed to
+    improve.
```

Comment:
No need to add this in the public documentation. You can put a comment when this attribute is defined for people reading the code however.

Reply:
Fixed in https://github.com/huggingface/transformers/pull/8581/commits/7f43217e908576bad51b05fc02202246ca3dfd4b